agents/ahv/fence_ahv.py

```diff
@@ -0,0 +1,744 @@
+"""
+AHV Fence agent
+
+Compatible with AHV v4 API
+"""
+
+
+import atexit
+import logging
+import sys
+import time
+import uuid
+import requests
+
+sys.path.append("@FENCEAGENTSLIBDIR@")
+import fencing
+
+
+V4_VERSION = '4.0'
+MIN_TIMEOUT = 60
+PC_PORT = 9440
+
+
+class NutanixClientException(Exception):
+        """
+        Exception class for Nutanix NutanixClient
+        """
+
+class AHVFenceAgentException(Exception):
+        """
+        Exception class for AHV Fence Agent
+        """
+
+
+class NutanixClient:
+        """
+        Nutanix REST Client
+        """
+        def __init__(self, username, password):
+                self.username = username
+                self.password = password
+                self.valid_status_codes = [200, 202]
+
+        def request(self, url, method='GET', headers=None, **kwargs):
+                """
+                Process requests based on the requested method.
+
+                Args:
+                url(str): URL of the API end point.
+                method(str): HTTP request method.
+                headers(dict): HTTP request headers
+                **kwargs: Any other keyword args relevant to the method.
+
+                Returns:
+                response: response object.
+                """
+                session = requests.Session()
+                session.auth = (self.username, self.password)
```

Comment:
same here! we can move both these line into init, if new session is not required for every request